src/main/java/frc/robot/subsystems/endEffector/EndEffector.java

```diff
@@ -10,13 +14,29 @@ public class EndEffector extends SubsystemBase {
   private final EndEffectorInputsAutoLogged inputs = new EndEffectorInputsAutoLogged();
   private final EndEffectorIO IO;
 
+  @AutoLogOutput(key = "EndEffector/hadAlgae")
+  private boolean hadAlgae = false;
+
   public EndEffector(EndEffectorIO IO) {
     this.IO = IO;
   }
 
   public void periodic() {
     IO.updateInputs(this.inputs);
     Logger.processInputs("EndEffector", inputs);
+
```

Comment:
[nit] There is a very very small scenario that calling hasAlgae() thrice in one iteration results in different return values each time. Might be safer to call once, store it, and then use it.